### ISSUE TITLE
Error after downloading a file using RxAlamofire

### ISSUE THREAD

**Issue by eniyavans** [RxSwiftCommunity/RxAlamofire#13]
Thanks for providing RxAlamofire.

Trying to download a zip file using  RxAlamofire

_ = manager!.rx_download (sourceURL ,destination:localPath)

```
        .map { downloadRequest ->Alamofire.Request in
            self.curentRequest = downloadRequest
            return downloadRequest
        }

        .flatMap { req -> Observable<(NSData?, RxProgress)> in
            let validatedRequest = req
                .validate(statusCode: 200 ..< 300)
                .validate(contentType: ["application/zip"])

            let data = req
                .rx_data()
                .map { responseData -> NSData? in responseData
               }
                .startWith(nil as NSData?)

            let progress = req.rx_progress()
            return Observable.combineLatest(data, progress) { ($0, $1) }
        }
```

Eventhough the zip file gets downloaded successfully, i am getting the alamofire error
“Data could not be serialized. Input data was nil."

Dependencies - Alamofire 3.1.4 ,RxAlamofire

Could you please provide an resolution for the above mentioned error?
Could you also provide an example for file download with progress using RxAlamofire?

**Comment by kremizask** [comment]
I am experiencing the same issue. Have you found a solution to this?

**Comment by sger** [comment]
@eniyavans @kremizask this is Alamofire issue check this old thread https://github.com/Alamofire/Alamofire/issues/233

**Comment by bontoJR** [comment]
Thanks for the reply @sger, do you have any idea on how to prevent this? I was able to reproduce the bug this morning, I would like to come up with a workaround or eventually state it as known issue in the doc.

**Comment by sger** [comment]
@bontoJR  writing a custom response handler but i think this is fixed in swift-2.0 branch of Alamofire
